docs/design/2022-06-06-Adding-Index-Accelerate.md

```diff
@@ -0,0 +1,117 @@
+
+---------------------------------------------------------------------------------------
+
+ 	[start1---end1 start2---end2 start3---end3 start4---end4  ...         ...         ]
+    |       |     |       |     |       |     |       |
+    +-------+     +-------+     +-------+     +-------+       ...         ...
+        |             |             |             |
+    bfworker1    bfworker2     bfworker3     bfworker4       ...         ...
+        |             |             |             |            |
+    localwriter1 localwriter2  localwriter3  localwriter4      |
+        +---------------- (round1)----------------+            +--(round2)--+
+                              |                                       |
+                      +------------------+                     +--------------
+                      |                  |                     |
+                 localfcworker1     localfcworker2        localfcwroker1     ...
+
+	|-----------------------------------------------------------------------------------|
+                                        |
+                                 +------------------+
+                                 |        |         |
+                             importer1 importer2  importer3     
+
+The new solution will set up a localwriter for each bfwoker, the building index key/value will be firstly writen into localwriter buffer and then be flushed to local storage at the end of every round. The localfcworker will do flush data and compaction tasks for building index data in TiDB Owner local. After all records are processed for index and local index has been done compaction(sorted that is why this solution need local storage for TiDB). The importer splits the local index sst file into small blocks and ingests them into TiKV. 
+
+### Advantage 
+1. Performance is good. Because the backfill processing is almostly done in TiDB local. The adding index performance is improved in an outstanding extend.
+2. Also since the backfill processing is not directly done on TiKV, theoretically it will impact less on user DML transaction. (Note: According limitations 3, The local CPU's consumption will contribute a negative part on this point. this will be solved in the future).
+      
+### Limitations
+As the first version, we have some limitations. in the feture, we will keep improve the solution.
+1. This solution will be an expirement feature released in TiDB 6.2.
+2. Checkpoint currently not support, that means we may restart the backfill process in TiDB DDL Owner restart scenario.
+3. The lightning need cpu and memory to make backfill faster, so currently it may consume more CPU. We have limited memory to be over used and will continue implement a flexible resource control framework for DDL.
+
+### Enable/Disable
+1. Use tidb_fast_ddl to enable and disable new solution.  
+```TiDB> set global tidb_fast_ddl = on/off;```
```

Comment:
```suggestion
```TiDB> set global tidb_enable_fast_ddl = on/off;```
```
This is existing the convention for boolean option flags.